docs: add initial guidelines on using AI. (#30)
OpenSTEF/.github: CONTRIBUTING.md

All patches and contributions, including those by project members, require review by one of the maintainers. We use GitHub pull requests for this purpose. Consult the [GitHub Help](https://help.github.com/articles/about-pull-requests/) for more information on pull requests.

## AI‑Assisted Contributions 
AI tools (e.g. LLMs or coding assistants) can be helpful when contributing to OpenSTEF. We welcome their use, as long as contributions remain clear, maintainable, and aligned with the following guidelines.

> **_NOTE:_**  These guidelines on using AI tools are still in development and will likely be extended over time.

### General expectations
- You remain fully responsible for your contribution, regardless of AI usage  
- You should understand and be able to explain the code you submit  
- Treat AI as a support tool, not as an autonomous contributor  

### Transparency
- If AI played a significant role, please disclose this in your pull request or commit message   (e.g. `Assisted-by: <tool name>`)  

> - Minor uses (e.g. fixing grammar or formatting) do not need to be mentioned  

Reviewer: I think it would be nice to be more clear about when mentioning is not necessary. Otherwise we create a grey area. We can start by only stating these two examples and extend if we find more cases where it does not make sense to mention that AI is used.
```suggestion
- If AI is only used for fixing grammar or formatting, it does not need to be mentioned
```

### Quality standards
AI-assisted contributions must meet the same standards as any other contribution:

- Include tests where applicable  
- Update documentation if needed  
- Follow coding style and project conventions (`poe all`)  
- Clearly describe the problem and your approach  

Please make sure to review and validate all generated code. AI tools can produce incorrect, outdated, or insecure patterns.

### Scope and collaboration
- Prefer small, well-scoped pull requests that are easy to review  
- Engage with maintainers via issues if your change is substantial  
- Avoid submitting large or bulk-generated changes without prior discussion, as they may be rejected.

### Security and licensing

> - Review generated code for potential vulnerabilities  

Reviewer: Idea (but not for this PR): could we not add a vulnerability gate that would not allow new dependencies if they are vulnerable?
Let me know what you think. I can make an issue if you agree.

Author: Agree! created issue https://github.com/OpenSTEF/openstef/issues/969
